src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -0,0 +1,140 @@
+      DBUtils.blockOnFuture(DBUtils.user(userId).delete());
+    }
+    
+    /** Add recipe id: true as a singleton map to user's created recipes field. */
+    public void addRecipeIdToCreated(String userId, String recipeId) {
+      DocumentReference userRef = DBUtils.user(userId);
+      String nestedPropertyName = DBUtils.getNestedPropertyName(User.CREATED_RECIPES_KEY, recipeId);
+      ApiFuture addToCreatedPostsFuture =
+          userRef.update(Collections.singletonMap(nestedPropertyName, true));
+      DBUtils.blockOnFuture(addToCreatedPostsFuture);
+    }
+    
+    /** Remove a recipe from a user's created (when recipe is deleted). */
+    public void removeRecipeIdFromCreated(String userId, String recipeId) {
+      DocumentReference userRef = DBUtils.user(userId);
+      String nestedPropertyName = DBUtils.getNestedPropertyName(User.CREATED_RECIPES_KEY, recipeId);
+      Map<String, Object> update = new HashMap<>();
+      update.put(nestedPropertyName, FieldValue.delete());
+      DBUtils.blockOnFuture(userRef.update(update));
+    }
+    
+    /** Add recipe id: true as a singleton map to user's saved recipes field. */
+    public void addRecipeIdToSaved(String userId, String recipeId) {
+      DocumentReference userRef = DBUtils.user(userId);
+      String nestedPropertyName = DBUtils.getNestedPropertyName(User.SAVED_RECIPES_KEY, recipeId);
+      ApiFuture addToSavedPostsFuture =
+          userRef.update(Collections.singletonMap(nestedPropertyName, true));
+      DBUtils.blockOnFuture(addToSavedPostsFuture);
+    }
+
+    /** Remove a recipe from a user's created (when recipe is deleted or unsaved). */
+    public void removeRecipeIdFromSaved(String userId, String recipeId) {
+      DocumentReference userRef = DBUtils.user(userId);
+      String nestedPropertyName = DBUtils.getNestedPropertyName(User.SAVED_RECIPES_KEY, recipeId);
+      Map<String, Object> update = new HashMap<>();
+      update.put(nestedPropertyName, FieldValue.delete());
+      DBUtils.blockOnFuture(userRef.update(update));
+    }
+
+    /** Add tag id: true as a singleton map to user's tags followed field. */
+    public void followTag(String userId, String tagId) {
+      DocumentReference userRef = DBUtils.user(userId);
+      String nestedPropertyName = DBUtils.getNestedPropertyName(User.TAGS_FOLLOWED_KEY, tagId);
+      ApiFuture addToTagsFollowedFuture =
+          userRef.update(Collections.singletonMap(nestedPropertyName, true));
+      DBUtils.blockOnFuture(addToTagsFollowedFuture);
+    }
+   
+    /** Remove a tag from a user's followed tags. */
+    public void unfollowTag(String userId, String tagId) {
+      DocumentReference userRef = DBUtils.user(userId);
+      String nestedPropertyName = DBUtils.getNestedPropertyName(User.TAGS_FOLLOWED_KEY, tagId);
+      Map<String, Object> update = new HashMap<>();
+      update.put(nestedPropertyName, FieldValue.delete());
+      DBUtils.blockOnFuture(userRef.update(update));
```

Comment:
This can all be super condensed using `userRef.update(nestedPropertyName , FieldValue.delete())`